Tweaks and fixes

ToeKnee opened the pull request:
I've created this as a single PR, but I'm happy to break it out into separate PRs, or you can cherry pick whatever you want.

This contains a bug fix, improved defaults on AMD and a couple of minor tweaks.

* 22ee5f2 - Fix  https://github.com/nishtahir/antec-flux-pro-display/issues/6
* 802e517 - On AMD CPUs default to Tctl for temperature
* dad8b34 - Increase default polling to 1000ms 
   This value was chosen because:
  * 200ms causes flickering on the display, making it hard to read.
  * `write_bulk` has a timeout of 1000ms, so we could be waiting on  `write_bulk` and sending more data at the same time.
  * On Intel CPUs, the temperature is an average of the past 256ms,  so we would be updating more often than the value.
* a7e5d1e - Version bump, copyright date and authors updates
* ca66770 - Update dependencies
* 2781d57 - Apply default clippy suggested fixes - I didn't add any non-standard clippy lints
* 003d14f -  Do not add the default CPU to config automatically. When restarting the Tctl device can change paths and is not consistant after a reboot.
  This commit changes:
  * Do not set the default CPU in the config automatically
  * Keep reading the CPU from the config if it exists
  * Look up the Tctl device when the service starts

ToeKnee committed:
Claim the USB interface to fix `Error writing bulk: Io`

ToeKnee committed:
Loop through hwmon devices looking for Tctl

On AMD CPUs Tctl is either, the actual CPU temperature, or the temperature with an offset.

On Threadripper CPUs, The offset version is used for cooling control, so is likely the one we want.
The actual temperature is under Tdie, which isn't available on most other Zen CPUS.

ToeKnee committed:
Increase default polling to 1000ms

This value was chosen becuase:

* 200ms causes flickering on the display, making it hard to read.
* `write_bulk` has a timeout of 1000ms, so we could be waiting on
  `write_bulk` and sending more data at the same time.
* On Intel CPUs, the temperature is an average of the past 256ms,
  so we would be updating more often than the value.

ToeKnee committed:
Version bump, copyright date update, add authors

ToeKnee committed:
Update dependencies

ToeKnee committed:
Apply clippy suggested fixes

nishtahir commented:
Ooh, thanks for the bug fixes. These are looking good. I don't mind the single PR to close these. I'd like to test the branch first before merging. I'll do that once I get to my desk later today

ToeKnee committed:
Do not add the default CPU to config automatically.

When restarting the Tctl device can change paths and is not consistant after a reboot.

This commit changes:
* Do not set the default CPU in the config automatically
* Keep reading the CPU from the config if it exists
* Look up the Tctl device when the service starts

ToeKnee committed:
When program is terminated, set temps to 0.0, 0.0

nishtahir commented:
Looks good to me. Thank you for your contribution. :tada:

ToeKnee commented:
> Looks good to me. Thank you for your contribution. 🎉

No problem!
